lib/Processor.php

```diff
@@ -841,29 +833,13 @@ public function initClosing() {
     }
 
     public function closeSocket() {
-        if ($this->readWatcher) {
-            Loop::cancel($this->readWatcher);
-        }
-        if ($this->writeWatcher) {
-            Loop::cancel($this->writeWatcher);
+        if ($this->socket) {
+            $this->socket->close();
         }
-        @\fclose($this->socket);
         $this->connectionState = self::CLOSED;
     }
 
-    private function compilePacket() {
-        do {
-            $pending = current($this->out);
-            unset($this->out[key($this->out)]);
-            if ($pending !== null || empty($this->out)) {
-                break;
-            }
-            $this->seqId = $this->compressionId = -1;
```

Comment:
the seqId/compressionId is not reset in the new code - that's likely causing bugs